foreign-key-mapping/README.md

```diff
@@ -0,0 +1,32 @@
+---
+layout: pattern
+title: Foreign-Key-Mapping
+folder: foreign-key-mapping
+permalink: /patterns/foreign-key-mapping/
+categories: Object-Relational Structural
+tags:
+- foreign-key
+---
+
+## Intent
+Maps an association between objects to a foreign key reference between tables.
+
```

Comment:
Add explanation section here. See the other patterns how it's implemented, for example https://github.com/iluwatar/java-design-patterns/tree/master/abstract-factory